internal/executor/rpc/cache.go

```diff
@@ -125,7 +126,17 @@ func (r *RPC) CacheInfo(ctx context.Context, req *api.CacheInfoRequest) (*api.Ca
 
 	r.logger.Debugf("sending info about cache key %s", req.CacheKey)
 
-	file, err := r.build.Cache.Get(req.CacheKey)
+	var file *os.File
+
+	if req.CacheKey != "" {
+		file, err = r.build.Cache.Get(req.CacheKey)
+	}
+	for _, prefix := range req.CacheKeyPrefixes {
+		file, err = r.build.Cache.FindByPrefix(prefix)
```

Comment:
There's a potential `*os.File` leak here, but given the ephemeral nature of `cirrus run`, we could probably just ignore it.

Reply:
Should be fixed by https://github.com/cirruslabs/cirrus-cli/pull/899.